model/syncPackage/migration/RdsMigrationService.php

```diff
@@ -0,0 +1,181 @@
+<?php
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2019  (original work) Open Assessment Technologies SA;
+ *
+ * @author Oleksandr Zagovorychev <zagovorichev@gmail.com>
+ */
+
+namespace oat\taoSyncClient\model\syncPackage\migration;
+
+
+use common_Logger;
+use common_persistence_Manager;
+use common_persistence_SqlPersistence;
+use Doctrine\DBAL\Query\QueryBuilder;
+use Doctrine\DBAL\Schema\SchemaException;
+use Doctrine\DBAL\Types\Type;
+use oat\oatbox\service\ConfigurableService;
+
+class RdsMigrationService extends ConfigurableService implements MigrationInterface
+{
+    const TABLE_NAME = 'sync_client_migrations';
+    const OPTION_PERSISTENCE = 'persistence';
+
+    public function __construct($options = array())
+    {
+        // if initialized within other service we need to rewrite config
+        if (!array_key_exists(self::OPTION_PERSISTENCE, $options)
+            && array_key_exists(0, $options) && count($options) === 1) {
+            $options = [self::OPTION_PERSISTENCE => current($options)];
+        }
+        parent::__construct($options);
+    }
+
+    public function getMigration($migrationId = 0)
+    {
+        $query = $this->getQueryBuilder()
+            ->select('*')
+            ->from(self::TABLE_NAME)
+            ->where(self::PARAM_SYNC_ID . ' = :syncId')
+            ->setParameter('syncId', $migrationId)
```

Comment:
It's a bit confusing what is `syncId` and what is `migrationId`

Reply:
one more time :)
**syncId** - unique value that shows us real process during which the data was sent to the server
**migrationId** - unique value that shows us the migration during which the data was collected to the package.